packages/pwa-kit-react-sdk/src/ssr/universal/components/route-component/index.js

```diff
@@ -406,7 +406,12 @@ export const routeComponent = (Wrapped, isPage, locals) => {
 export const getAllRoutes = async (locals = {}) => {
     const {applicationExtensions = []} = locals
     const extensionRoutes = (
-        await Promise.all(applicationExtensions.map((extension) => extension.getRoutes()))
+        await Promise.all(applicationExtensions.map((extension) => {
+            if (extension.getRoutesAsync) {
+                return extension.getRoutesAsync()
+            }
+            return extension.getRoutes()
+        }))
```

Comment:
Minor: more concise version

```suggestion
        await Promise.all(applicationExtensions.map((extension) => {
            return extension.getRoutesAsync ? extension.getRoutesAsync() : extension.getRoutes()
        }))
```